.github/workflows/e2e_tests.yml

```
@@ -41,29 +27,28 @@ jobs:
       - name: Setup pnpm
         uses: ./.github/actions/pnpm-setup
```

ask: 

is this kind of `if:` needed?

```suggestion
      - name: Setup pnpm
        uses: ./.github/actions/pnpm-setup
        if: |
          github.event_name == 'deployment_status' &&
          github.event.deployment_status.state == 'success' &&
          (
            (
              contains(github.event.deployment.environment, 'Production') &&
              contains(github.event.deployment_status.target_url, 'liam-app-git-main')
            ) ||
            (
              contains(github.event.deployment.environment, 'Preview') &&
              !contains(github.event.deployment_status.target_url, 'liam-erd-sample') &&
              !contains(github.event.deployment_status.target_url, 'liam-docs')
            )
          )
```

---

As you say, we need ``if`` in every step.🙏
I have added a step called “Check deployment conditions” so that the following code can be used to determine this.
```
if: steps.check.outputs.should_run == 'true'
```
https://github.com/liam-hq/liam/pull/1053/commits/fd7746e3e87e0258bf18bbc4b3ebaef23b0bd56b